mlflow/gateway/config.py

```diff
@@ -44,18 +44,65 @@ def validate_api_key(cls, value):
         return _resolve_api_key_from_input(value)
 
 
+class OpenAIAPIType(str, Enum):
+    OPENAI = "openai"
+    AZURE = "azure"
+    AZUREAD = "azuread"
+
+
 class OpenAIConfig(BaseModel, extra=Extra.allow):
     openai_api_key: str
-    openai_api_type: Optional[str] = None
-    openai_api_base: str = "https://api.openai.com/v1"
+    openai_api_type: OpenAIAPIType = OpenAIAPIType.OPENAI
+    openai_api_base: Optional[str] = None
     openai_api_version: Optional[str] = None
+    openai_deployment_name: Optional[str] = None
     openai_organization: Optional[str] = None
 
     # pylint: disable=no-self-argument
     @validator("openai_api_key", pre=True)
     def validate_openai_api_key(cls, value):
         return _resolve_api_key_from_input(value)
 
+    # pylint: disable=no-self-argument
+    @validator("openai_api_type", pre=True)
+    def validate_openai_api_type(cls, value):
+        api_type = value.lower()
+        if api_type not in [api_type.value for api_type in OpenAIAPIType]:
+            raise MlflowException.invalid_parameter_value(f"Invalid OpenAI API type '{api_type}'")
+        return api_type
```

Comment:
Do we still need this? Does pydantic automatically validate this field?

Reply:
Yes, we still need it if we want case-insensitive api type parsing (`lower()`)

Reply:
@harupy I implemented `_missing` (https://docs.python.org/3.11/howto/enum.html#supported-sunder-names) to address this and removed the extra validation logic. Looks like `_missing` has been around since Python 3.6, so we should be okay :D. pydantic is very clever!